src/main/java/com/openlattice/datastore/data/controllers/DataIntegrationController.java

```diff
@@ -216,7 +219,40 @@ public IntegrationResults integrateEntityAndAssociationData(
     @Override
     @PutMapping( "/" + EDGES )
     public int createEdges( @RequestBody Set<DataEdgeKey> edges ) {
-        return dgm.createAssociations( edges ).getNumUpdates();
+        final var associationEntitySetIds = new HashMap<UUID, Map<UUID, Set<UUID>>>(); // edge-src-dst
+
+        edges.forEach(
+                association -> {
+                    final var associationEntitySetId = association.getEdge().getEntitySetId();
+                    final var srcEntitySetId = association.getSrc().getEntitySetId();
+                    final var dstEntitySetId = association.getDst().getEntitySetId();
+
+                    //Ensure that we have read access to entity set metadata.
+                    ensureReadAccess( new AclKey( srcEntitySetId ) );
+                    ensureReadAccess( new AclKey( dstEntitySetId ) );
+                    ensureReadAccess( new AclKey( associationEntitySetId ) );
```

Comment:
If this is called during an integration this might be duplicating these calls tens of thousands of times -- would probably be faster to collect a map of all entity sets written -> all property types written on those entity sets, and then do the access checks once